src/pages/SingleCell/PieToolTip.tsx

```diff
@@ -0,0 +1,393 @@
+    // const [isHovered, setIsHovered] = useState<boolean>(false);
+    const {
+        pieChartData,
+        tooltipEnabled,
+        downloadSvg,
+        setDownloadSvg,
+        downloadPdf,
+        setDownloadPdf,
+        heading,
+        isHovered,
+        setIsHovered,
+        hoveredSliceIndex,
+        setHoveredSliceIndex,
+    } = singleCellStore;
+
+    console.log('Current hoveredSliceIndex:', hoveredSliceIndex);
+    const [tooltipVisible, setTooltipVisible] = useState<boolean>(false);
+    const [tooltipHovered, setTooltipHovered] = useState<boolean>(false);
+    const [downloadOptionsVisible, setDownloadOptionsVisible] = useState<
+        boolean
+    >(false);
+
+    // Create a ref to hold the SVG container
+    const svgRef = useRef<SVGSVGElement>(null);
+    // Set to store unique patient IDs
+    let differentPatientIds: string[] = [];
+
+    // Extract unique patient IDs from piechartData
+    for (let i = 0; i < pieChartData.length; i++) {
+        let currentId = pieChartData[i].patientId;
+        if (!differentPatientIds.includes(currentId)) {
+            differentPatientIds.push(currentId);
+        }
+    }
+    // Initialize an object to store data for each patient
+    // let patientData = {};
+    let patientData: PatientData = {};
+
+    // Iterate over unique patient IDs
+    for (let i = 0; i < differentPatientIds.length; i++) {
+        let id = differentPatientIds[i];
+        patientData[id] = []; // Initialize array for current patient ID
+
+        // Iterate over piechartData to find data for current patient
+        for (let j = 0; j < pieChartData.length; j++) {
+            if (pieChartData[j].patientId === id) {
+                patientData[id].push({
+                    stableId: pieChartData[j].stableId,
+                    value: pieChartData[j].value,
+                });
+            }
+        }
+    }
+    useEffect(() => {
+        if (downloadSvg) {
+            handleDownloadSVG(svgRef);
+            setDownloadSvg(false);
+        }
+    }, [downloadSvg]);
+
+    useEffect(() => {
+        if (downloadPdf) {
+            handleDownloadPDF(svgRef);
+            setDownloadPdf(false);
+        }
+    }, [downloadPdf]);
+
+    // Set tooltip visibility with a delay when hover state changes, only if tooltipEnabled is false
+    useEffect(() => {
+        if (tooltipEnabled) {
+            setTooltipVisible(true);
+        } else {
+            if (isHovered || tooltipHovered) {
+                setTooltipVisible(true);
+            } else {
+                const timeoutId = setTimeout(
+                    () => setTooltipVisible(false),
+                    300
+                ); // 300ms delay before hiding tooltip
+                return () => clearTimeout(timeoutId);
+            }
+        }
+    }, [isHovered, tooltipHovered, tooltipEnabled]);
+
+    // Define color scale (replace with your desired colors)
+    const colors = [

```

Comment:
I think you are defining the same colors in multiple files. You could create a file `SingeCellPageUtils`, define the colors there and import them when needed.

Reply:
Yes, I have defined the colors in singleCellStore and used them by importing from there.